src/dbnode/storage/index.go

```diff
@@ -1092,10 +1074,17 @@ func (i *nsIndex) WarmFlush(
 				zap.Error(err),
 				zap.Time("blockStart", block.StartTime().ToTime()),
 			)
+		} else {
+			for _, s := range shards {
+				shardFlushes[shardFlush{
+					shard: s,
+					time:  block.StartTime(),
+				}] = true
+			}
```

Comment:
Same here, would avoid the else with a continue above (as per Effective Go):
```suggestion
			continue
		}
		
		for _, s := range shards {
			shardFlushes[shardFlush{
				shard: s,
				time:  block.StartTime(),
			}] = true
		}
```